File: tests/fixtures/nestjs-comprehensive/src/orders/order.controller.ts
import { Controller, Get, Post, Patch, Body, Param } from "@nestjs/common";
import { CreateOrderDto, UpdateOrderStatusDto } from "./order.dto";

@Controller("orders")
export class OrderController {
  @Get()
  list() {
    return [];
  }

  @Post()
  create(@Body() body: CreateOrderDto) {
    return { id: 1, ...body };
  }

  @Get(":id")
  show(@Param("id") id: string) {
    return { id };
  }

  @Patch(":id/status")
  updateStatus(@Param("id") id: string, @Body() body: UpdateOrderStatusDto) {
    return { id, ...body };
  }
}

File: tests/fixtures/springboot-comprehensive/src/main/java/com/example/orders/OrderController.java
package com.example.orders;

import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/orders")
public class OrderController {

    @GetMapping
    public List<Order> list() { return List.of(); }

    @PostMapping
    public Order create(@RequestBody @Valid Order body) { return body; }

    @GetMapping("/{id}")
    public Order show(@PathVariable String id) { return new Order(); }

    @PatchMapping("/{id}/status")
    public Order updateStatus(@PathVariable String id, @RequestBody @Valid Order body) { return body; }
}